src/rules/quotePropsRule.ts

```diff
@@ -0,0 +1,103 @@
+import * as Lint from "../lint";
+import * as ts from "typescript";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "quote-props",
+        description: "Quoting Style for Property Names",
+        descriptionDetails: Lint.Utils.dedent`
+            Object literal property names can be defined in two ways: using literals or using strings.
+            For example, these two objects are equivalent:
+
+            var object1 = {
+                property: true
+            };
+
+            var object2 = {
+                "property": true
+            };
+
+            In many cases, it doesn’t matter if you choose to use an identifier instead of a string
+            or vice-versa. Even so, you might decide to enforce a consistent style in your code.
+
+            This rules lets you enforce consistent quoting of property names. Either they should always
+            be quoted (default behavior) or quoted only as needed ("as-needed").
+        `,
+        optionsDescription: Lint.Utils.dedent`
+            Possible settings are:
+
+            * \`"always"\`: Property names should always be quoted. (This is the default.)
+            * \`"as-needed"\`: Only property names which require quotes may be quoted (e.g. those with spaces in them).
+
+            For ES6, computed property names (\`{[name]: value}\`) and methods (\`{foo() {}}\`) never need
+            to be quoted.
+        `,
+        options: {
+            type: "array",
+            items: [
+                {
+                    enum: ["always", "as-needed"],
+                    // TODO: eslint also supports "consistent", "consistent-as-needed" modes.
+                    // TODO: eslint supports "keywords", "unnecessary" and "numbers" options.
+                },
+            ],
+            minItems: 0,
+            maxItems: 1,
+        },
+        optionExamples: ["true", "[true, \"as-needed\"]", "[true, \"always\"]"],
```

Comment:
Once we start validating rule options, we'll probably require specifying the mode for the rule. I'd rather that in the documentation we don't mention a default value